### ISSUE TITLE
Configure the ha_group parameter of na_sg_grid_gateway to accept IDs of HA groups

### ISSUE THREAD

**Issue by Harppi** [ansible-collections/netapp.storagegrid#10]
##### SUMMARY

Improve the flexibility of `na_sg_grid_gateway` by configuring its `ha_group` parameter to accept IDs of HA groups.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

na_sg_grid_gateway

##### ADDITIONAL INFORMATION

The [na_sg_grid_gateway.py](https://github.com/ansible-collections/netapp.storagegrid/blob/main/plugins/modules/na_sg_grid_gateway.py) module requires the `ha_groups` parameter to define the names of HA groups instead of their IDs. It would be straightforward to write an Ansible role that determines the names, IDs and HA groups of load balancer endpoints on its own without hard coding anything if the module accepted the IDs of HA groups. This would also match the behavior of the `/private/gateway-configs` operation.

Currently, running a GET request against `/private/gateway-configs` returns the names and IDs of load balancer endpoints and the IDs of their HA groups, and running a GET request against `/private/ha-groups` returns the names and IDs of HA groups. While these two operations return all the necessary information, it still requires some further work to combine the information and convert the IDs of HA groups to the corresponding names within the JSON object returned by `/private/gateway-configs` operation. The conversion has proven to be somewhat tricky if it must be done programmatically and without hard coding the names of HA groups.

In other words, `/private/gateway-configs` lacks the names of HA groups and `/private/ha-groups` lacks the ID of the load balancer endpoint to which an HA group is bound. `Nodes > <node_name> > Load Balancer` in the Grid Manager apparently presents the binding.

The following snippet demonstrates uploading a renewed certificate to multiple load balancer endpoints. Previous tasks of an Ansible role have already defined the following variables:

* `auth.json.data` - a Grid Management API token
* `grid_admin_base_url` - the URL of a primary admin node
* `endpoint_cert` - a base64 encoded server certificate
* `endpoint_chain` - a base64 encoded CA bundle
* `endpoint_privkey` - a base64 encoded certificate private key

```yaml
- name: Get load balancer endpoint configurations
  uri:
    url: "{{ grid_admin_base_url }}/api/v3/private/gateway-configs"
    method: GET
    headers:
      accept: application/json
      Authorization: Bearer {{ auth.json.data }}
    validate_certs: false  # skip validation to renew expired certificates
  register: gateways

# Create a nested list of display names, IDs and HA groups
# [["displayName1", "id1", ["haGroups1"]], ["displayName2", "id2", ["haGroups2"]]]
- name: Map display name to ID and HA groups
  set_fact:
    gateways: "{{ gateways | json_query(query) }}"
  vars:
    query: "json.data[*].[displayName, id, pinTargets.haGroups]"
    
- name: Deploy load balancer endpoint certificate
  netapp.storagegrid.na_sg_grid_gateway:
    api_url: "{{ grid_admin_base_url }}"
    auth_token: "{{ auth.json.data }}"
    validate_certs: false  # skip validation to renew expired certificates
    state: present
    display_name: "{{ item[0] }}"  # see the structure of the nested list above
    gateway_id: "{{ item[1] }}"  # see the structure of the nested list above
    port: 10443
    secure: true  # only serve HTTPS traffic
    enable_ipv4: true
    enable_ipv6: true
    binding_mode: ha-groups
    ha_groups: "{{ item[2] }}"   # see the structure of the nested list above
    default_service_type: "s3"
    server_certificate: "{{ endpoint_cert['content'] | b64decode }}"
    private_key: "{{ endpoint_privkey['content'] | b64decode }}"
    ca_bundle: "{{ endpoint_chain['content'] | b64decode }}"
  no_log: true  # hide cert, chain and privkey
  loop: "{{ gateways }}"
```

**Comment by joshedmonds** [comment]
Resolved in 789ec0af371d5e4be75d314ca7eca3e3da47d800

**Comment by joshedmonds** [comment]
Available in Ansible Galaxy release 21.10.0.
